moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -457,17 +536,24 @@ bool JointModelGroup::getVariableDefaultPositions(const std::string& name, std::
 
 void JointModelGroup::getVariableDefaultPositions(double* values) const
 {
-  for (std::size_t i = 0; i < active_joint_model_vector_.size(); ++i)
-    active_joint_model_vector_[i]->getVariableDefaultPositions(values + active_joint_model_start_index_[i]);
+  for (const auto& [i, joint_model] : active_joint_model_vector_ | views::enumerate)
+  {
+    joint_model->getVariableDefaultPositions(values + active_joint_model_start_index_[i]);
+  }
   updateMimicJoints(values);
 }
 
 void JointModelGroup::getVariableDefaultPositions(std::map<std::string, double>& values) const
 {
-  std::vector<double> tmp(variable_count_);
-  getVariableDefaultPositions(&tmp[0]);
-  for (std::size_t i = 0; i < variable_names_.size(); ++i)
-    values[variable_names_[i]] = tmp[i];
+  const auto default_positions = [=]() {
+    auto tmp = std::vector<double>(variable_count_);
+    getVariableDefaultPositions(&tmp[0]);
```

Comment:
I wonder what happens if the variable counts ends up being zero... The same problem is of course also present in the current code.

Reply:
I don't know... that's a good question.  I wasn't trying to change the behavior of existing code though with this PR.